gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/utils/EhrResourceExtractorUtil.java

```diff
@@ -0,0 +1,47 @@
+package uk.nhs.adaptors.pss.translator.utils;
+
+import org.hl7.v3.II;
+import org.hl7.v3.RCMRMT030101UK04Component;
+import org.hl7.v3.RCMRMT030101UK04Component3;
+import org.hl7.v3.RCMRMT030101UK04Component4;
+import org.hl7.v3.RCMRMT030101UK04EhrComposition;
+import org.hl7.v3.RCMRMT030101UK04EhrExtract;
+import org.hl7.v3.RCMRMT030101UK04EhrFolder;
+
+import java.util.List;
+
+public class EhrResourceExtractorUtil {
+
+    public static RCMRMT030101UK04EhrComposition extractEhrCompositionForPlanStatement(RCMRMT030101UK04EhrExtract ehrExtract,
+                                                                                       II resourceId) {
+        return ehrExtract.getComponent()
+                .stream()
```

Comment:
One indent here as well please :) And `.filter(component -> hasEhrFolder(component))` and `.filter(component -> hasEhrComposition(component))` could be replaced with method references